### ISSUE TITLE
discussion: a metrics plugin using github.com/VictoriaMetrics/metrics

### ISSUE THREAD

**Issue by vrischmann** [twmb/franz-go#875]
Hi,

this is not really an issue, I'm opening a discussion.

We use [VictoriaMetrics](https://victoriametrics.com) at work and its [metrics](https://github.com/VictoriaMetrics/metrics) library to expose metrics from Go programs. It's a lightweight alternative to the official prometheus client.
It's not necessary to use VictoriaMetrics, it's just another, more efficient way to expose metrics in the Prometheus text format (and its histogram is [easier to use](https://valyala.medium.com/improving-histogram-usability-for-prometheus-and-grafana-bc7e5df0e350)).

I'm wondering if you would be open to include a plugin like kvictoria or something that would expose metrics using the `metrics` library, that way we could benefit from the advantages of the `metrics` library.

I'm willing to work on this. My plan would be to essentially copy and adapt the kprom plugin.

Let me know what you think.

**Comment by vtolstov** [comment]
[metrics](https://github.com/VictoriaMetrics/metrics) library does not suffort prometheus histograms

**Comment by vrischmann** [comment]
> [metrics](https://github.com/VictoriaMetrics/metrics) library does not support prometheus histograms

Do you mean that Prometheus (the database) wouldn't know what to do with a Histogram from the metrics library ?
That's probably true but this plugin would be for users of VictoriaMetrics (the database).

**Comment by vtolstov** [comment]
i'm use metrics library in some of my projects, but without prometheus like histograms metrics can't be collected by prometheus server. So this plugin can be useful only with victoriametrics server that collects metrcis

**Comment by vrischmann** [comment]
Yes. That's my use case and it might also benefit other users of VictoriaMetrics server so I'd argue a plugin is still useful.

**Comment by vtolstov** [comment]
nice to hear!

**Comment by twmb** [comment]
I'm open to this, and will review/tag a release if you open a PR. Please let me know in the PR message if this involves changes to kgo because that'll need to be merged/tagged independently.

**Comment by vrischmann** [comment]
Hi, thanks for the feedback. 

I've started working on this and I realized that I'll have the same problem that is addressed in #820, in fact I've had this problem at work because we sometimes use multiple clients per application.

As far as I can tell the changes in #820 will work for me too.

**Comment by twmb** [comment]
Released as plugin/kvictoria/v1.0.0
